tests/containers/podman_pods.pm

```diff
@@ -18,6 +18,43 @@ use serial_terminal 'select_serial_terminal';
 use version_utils qw(is_sle is_opensuse is_sle_micro);
 use containers::k8s qw(install_k3s uninstall_k3s);
 
+sub check_container_nspid {
+    # expect set hostpid
+    my $hostpid = shift;
+    my $config = 'hello-kubic.yaml';
+    my @errors = ();
+
+    record_info('NSPid', sprintf('Expect HostPid == %s', !!$hostpid ? 'True' : 'False'));
+
+    if ($hostpid) {
+        assert_script_run(qq[sed -i '/containers:/i\\      hostPID: true' $config]);
+    }
+
+    assert_script_run("podman play kube --replace $config");
+    my $pod = script_output(q[podman pod ps --format '{{ .Name }}']);
+    my $container = script_output(q[podman container ps --filter name=.*pod.* --format '{{ .Names }}']);
+    my $pid_on_host = script_output(qq[podman container inspect $container --format '{{ .State.Pid }}']);
```

Comment:
Although it's fetching the Pid of the running container, for some reason, I'm able to see different PIDs on multiple runs of this command. 

https://github.com/containers/podman/blob/832b098471d7abb19fbfe80bf1a97bfa243ff749/libpod/container.go#L163-L164

I'd additionally suggest we can try `podman exec <container_name> echo $$`, which should never be 1 _if_ hostPID is set to true.